/*
 * Copyright 2018 The Feast Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package feast.specs;

option java_package = "feast.specs";
option java_outer_classname = "ImportSpecProto";
option go_package = "github.com/gojek/feast/protos/generated/go/feast/specs";

import "google/protobuf/timestamp.proto";

message ImportSpec {
  // type should define what sort of store it is
  // pubsub: topic, subscription (already a featureRow)
  // kafka: ? (already a featureRow)
  // file: csv, parquet, json
  // bigquery: table, query
  string type = 1;

  map<string, string> options = 2;

  repeated string entities = 3;

  Schema schema = 4;
}

message Schema {
  repeated Field fields = 1;
  oneof timestamp { // the event timestamp to set per row.
    string timestampColumn = 5;
    google.protobuf.Timestamp timestampValue = 6;
  }
  string entityIdColumn = 7;
}

message Field {
  string name = 1;
  string featureId = 2; // set this if this is a feature
}
